### PR TITLE
Polish forms components and split out CSS files

### PR DESCRIPTION
**Pull request by janfaracik**
This PR is the first of several follow ups to #5923, the intention of this PR is to polish the new form components and to also modularise the styling of the components.

* Form components have been polished, icons have been updated and animations have been tweaked to be more consistent

Kapture 2022-02-07 at 19 56 58

* CodeMirror input has been updated to look the same as textarea

image

* Component styling has been split out into individual files to make it easier to maintain
* Some small spacing adjustments + headings have been updated to be the same size as `.jenkins-app-bar` without the need for the additional wrapping

### Proposed changelog entries

* Minor form and spacing changes.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: war/src/main/less/base/style.less
Comment (timja) [inline war/src/main/less/base/style.less:228]: is this intentional?

image

Comment (timja) [reply war/src/main/less/base/style.less:228]: Looks nice on dark theme:

image

Comment (NotMyFault) [reply war/src/main/less/base/style.less:228]: Looks quite nice on dark theme for console output, but applying it on every `<pre>` looks weird:

Comment (timja) [reply war/src/main/less/base/style.less:228]: May be okay on dark theme there, e.g script console page looks nice

**Review by NotMyFault — COMMENTED** [review]
- Amending to my other comment [here](https://github.com/jenkinsci/jenkins/pull/6259#discussion_r801079845)

Changing the theme for console highlights looks quite nice in dark theme, but a bit odd in other occasions, such as `/log/all` for example.

- The scripting console's expandable textarea drag button does nothing, though it wasn't expandable in the first place

Maybe move the "Run" button a bit lower, so it doesn't cut off the new text box frame too soon?

- Unless that is a regression from the form components PR, you can't change the size of the textboxes at `/editDescription`, whether job or Jenkins' description, anymore, if you attempt to drag it smaller or larger.

File: war/src/main/less/base-styles-v2.less
Comment (fqueiruga) [inline war/src/main/less/base-styles-v2.less:37]: LOVE all of this

**Review by fqueiruga — APPROVED** [review]
Code looks good, haven't tried it. This is some great refactoring!

**Review by timja — APPROVED** [review]
Only a couple of nits but non blocking,

1. script console looks like it can be resized but it doesn't let you
2. dark theme should probably have the pre-background variable set to increase the contrast on the console log

**Review by NotMyFault — APPROVED** [review]
>Only a couple of nits but non blocking,
script console looks like it can be resized but it doesn't let you

Yeah, that actually is a regression from the form components PR that can be addressed separately I guess. Same applies to the job description text box, but Jan already took care of that 👀 

The new `<pre>`s look much better imo, only nit I have are, that links now use the same color as regular text, used to be blue before.

**Comment by janfaracik** [comment]
> > Only a couple of nits but non blocking,
> > script console looks like it can be resized but it doesn't let you
> 
> Yeah, that actually is a regression from the form components PR that can be addressed separately I guess. Same applies to the job description text box, but Jan already took care of that 👀
> 
> The new `\<pre\>`s look much better imo, only nit I have are, that links now use the same color as regular text, used to be blue before.

I've fixed the link colour in `<pre/>` 👍 The resize issue is really weird, works in Safari but not other browsers - will need a good Google to find if others have a similar problem with CodeMirror ...

> dark theme should probably have the pre-background variable set to increase the contrast on the console log

I'll update the theme 👍

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
Contributed to [JENKINS-69549](https://issues.jenkins.io/browse/JENKINS-69549).

File: war/src/main/less/base/typography.less
Comment (basil) [inline war/src/main/less/base/typography.less:55]: @janfaracik This change seems intended to fix some particular problem, but it's unclear which problem it was intended to fix and in fact it causes a new problem as described in JENKINS-69549 and #7078. In order to ensure that the fix for that bug doesn't re-expose whatever problem this original change was intended to fix, it would be helpful to know what specifically this original change was intended to fix.

**Pull request merged**
